research-software-directory/RSD-as-a-service
# 1212 categories for communities

## Categories for communities, interface for creating and updating categories

Changes proposed in this pull request:

* Reorder database initialisation files to allow categories for communities
* Added size limits on various category columns
* Add frontend interface for admins to maintain global categories
* Add frontend interface for maintainers of a community to maintain their categories
* The data generation module now creates categories

How to test:

* Make sure to have communities enabled
* `docker compose down --volumes && docker compose build --parallel && docker compose up --scale data-generation=1`
* Sign in as admin
* Go to the admin pages -> categories
* Play around with the categories (create new, edit, delete)
* Invite a maintainer for a community, sign in with that account
* Play around with the categories (create new, edit, delete) on the community page
* **Important:** check in the SQL files that nothing got lost unintentionally

**Todo in this PR**:

* [x] On the software edit page, only show global categories

**Todo:**

* Allow icons to be added to categories (see #975)
* Force software page maintainer to add categories when they want to join a community (see #1212)


Related to #1216

Related to #1212

Related to #1141

Related to #992

Related to #975

PR Checklist:
